conf/template/icub-config-build-tree.cmake.in

```diff
@@ -4,21 +4,30 @@
 
 if (NOT ICUB_FOUND)
 
-message(STATUS "Using iCub from build tree")
+  message(STATUS "Using iCub from build tree")
 
-set(ICUB_VERSION @ICUB_VERSION@)
+  set(ICUB_VERSION @ICUB_VERSION@)
+  # If YARP_LIBRARIES has been valorized by a previous call of find_package(YARP)
+  # in user CMakelists.txt save it in temporary variable to restore it later.
+  if(YARP_LIBRARIES)
```

Comment:
I would change this to `if(DEFINED YARP_LIBRARIES)`, but in this case the effect is exactly the same.
By the way, @traversaro do you think that it is correct that `YARP_LIBRARIES` is overwritten every time `find_package(YARP) is called, or perhaps they should be appended?

Reply:
> By the way, @traversaro do you think that it is correct that `YARP_LIBRARIES` is overwritten every time `find_package(YARP) is called, or perhaps they should be appended?

To be honest, I am never seen any example of a variable only containing the targets specified in the components. Having said that, indeed the fact that any build system that use `YARP_LIBRARIES` can be broken by a random call to `find_package(YARP)` (even in a nested and hidden file!) seems to be extremely dangerous, so perhaps appending to `YARP_LIBRARIES` may be the lesser evil.

Reply:
Keeping `YARP_LIBRARIES` with the old meaning or deprecating it seems to me equally good solutions (modulo the massive amount of change necessary to do if `YARP_LIBRARIES` if it is removed) w.r.t. to this problem.

Reply:
Please, I want to have a break from the awful task of updating our user code. Let's solve for now any problem related to `YARP_LIBRARIES` at this level.